packages/flutter/test/widgets/basic_test.dart

```diff
@@ -278,6 +279,90 @@ void main() {
       equals('UnconstrainedBox(alignment: topRight, constrainedAxis: horizontal, textDirection: rtl)'),
     );
   });
+
+  group('ColoredBox', () {
+    _MockCanvas mockCanvas;
+    _MockPaintingContext mockContext;
+    const Color colorToPaint = Color(0xFFABCDEF);
+
+    setUp(() {
+      mockContext = _MockPaintingContext();
+      mockCanvas = _MockCanvas();
+      when(mockContext.canvas).thenReturn(mockCanvas);
+    });
+
+    testWidgets('ColoredBox - no size, no child', (WidgetTester tester) async {
+      await tester.pumpWidget(Flex(
+        direction: Axis.horizontal,
+        textDirection: TextDirection.ltr,
+        children: const <Widget>[
+          SizedBox.shrink(
+            child: ColoredBox(color: colorToPaint),
+          ),
+        ],
+      ));
+      expect(find.byType(ColoredBox), findsOneWidget);
+      final RenderObject renderColoredBox = tester.renderObject(find.byType(ColoredBox));
+
+      renderColoredBox.paint(mockContext, Offset.zero);
+
+      verifyNever(mockCanvas.drawRect(any, any));
+      verifyNever(mockContext.paintChild(any, any));
+    });
+
+    testWidgets('ColoredBox - no size, child', (WidgetTester tester) async {
+      const ValueKey<int> key = ValueKey<int>(0);
+      const Widget child = SizedBox.expand(key: key);
+      await tester.pumpWidget(Flex(
+        direction: Axis.horizontal,
+        textDirection: TextDirection.ltr,
+        children: const <Widget>[
+          SizedBox.shrink(
+            child: ColoredBox(color: colorToPaint, child: child),
```

Comment:
Does the `child = SizedBox.expand(key: key)` also have 0 size? My first reaction is that a parent `ColoredBox` is at least the size of its child, so a 0-sized parent `ColoredBox` infers a 0-sized child.

Reply:
We have to break out of the constraints of the tester to make any of this work - we could also use a child instead of a parent to do it, but we need to make sure we're in a context where we won't just take up the full available space anwyya (e.g. using a flexible).